### PR TITLE
Don't create credentials request if the secret is given in CR

### PR DESCRIPTION
**Pull request by alebedev87**

### CODE REVIEW

**Comment by alebedev87** [comment]
AWS test timed out, the set of received IPs was a bigger than the expected:
```
    operator_test.go:300: Getting IPs of service's load balancer
    operator_test.go:333: Loadbalancer's IP(s): map[34.215.81.125:{}]
    operator_test.go:341: Looking for DNS record in nameserver: ns-1583.awsdns-05.co.uk
    operator_test.go:351: Got IPs: [34.215.81.125 54.187.28.21]
```

**Comment by alebedev87** [comment]
/assign @sherine-k 

This PR is what you noticed in one of your tests.

File: test/e2e/operator_test.go [hunk 1]
```diff
@@ -348,13 +348,20 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
...
 			}
-			for _, ip := range ips {
-				if _, ok := serviceIPs[ip]; !ok {
+			// all expected IPs should be in the received IPs
+			// but these 2 sets are not necessary equal
+			for ip := range serviceIPs {
+				if _, found := gotIPs[ip]; !found {
 					return false, nil
 				}
 			}
```
Comment (alebedev87) [inline test/e2e/operator_test.go:367]: @arjunrn: this is to fix [this issue](https://github.com/openshift/external-dns-operator/pull/104#issuecomment-995174939). It doesn't happen all the time but when it does, the test times out.

**Review by sherine-k — APPROVED** [review]
Thanks @alebedev87 !

File: test/e2e/operator_test.go (hunk 1 above)
Comment (arjunrn) [reply test/e2e/operator_test.go:367]: I would like to see this change in a separate PR if possible. Also `gotIPs` could be called `loadbalancerIPs`.

**Pull request merged**